### PR TITLE
WIP: DOC: add a SciPy Code of Conduct.

### PR DESCRIPTION
**Pull request by rgommers**
*Main discussion for this on the mailing list*.

To be done:

- [ ] decide what email address to use (conduct@scipy.org may be hard to set up or taken by the SciPy conference).
- [ ] move this from Markdown to reST and integrate with the rest of the docs
- [ ] add a top-level ``CODE_OF_CONDUCT`` file which links to this content

### CODE REVIEW

File: doc/source/dev/conduct/enforcement.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+(vetted by all parties to the process) to the Committee, with recommendations
+on further steps.  The Committee will then evaluate these results (whether
+satisfactory resolution was achieved or not) and decide on any additional
+action deemed necessary.
+
+
+## How the committee will respond to reports
+
+When a report is sent to the committee they will immediately reply to the
+reporter to confirm receipt. This reply must be sent within 24 hours, and the
```
Comment (person142) [inline doc/source/dev/conduct/enforcement.md]: This should probably also be changed to 72 hours?

File: doc/source/dev/conduct/code_of_conduct.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+a timely manner. Code of conduct violations reduce the value of the community
+for everyone and we take them seriously.
+
+You can file a report by emailing *TODO: email address* or alternatively by contacting
+the NumFOCUS reporting contact (https://www.numfocus.org/about/code-of-conduct/).
+For more details, please see our [Reporting Guidelines](link).
+
+You can request that we follow up with you directly.  If you prefer to keep your
+report anonymous, please send it from an email account that isn't linked to your
+identity in a manner that can be found publicly.  While we cannot follow up on
```
Comment (person142) [inline doc/source/dev/conduct/code_of_conduct.md]: Might read better if it says "While we cannot follow up with you directly on an anonymous report, we will take appropriate action." Otherwise it reads like we will and won't take action.

Comment (rgommers) [inline doc/source/dev/conduct/code_of_conduct.md]: thanks, fixed

File: doc/source/dev/conduct/enforcement.md (hunk 1 above)
Comment (rgommers) [inline doc/source/dev/conduct/enforcement.md]: yep, good catch

File: doc/source/dev/conduct/enforcement.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+and fair.
+
+## The Code of Conduct Committee
+
+All responses to reports of conduct violations will be managed by a Code of
+Conduct Committee ("the committee"), currently comprised of
+[*note:* still gathering volunteers, this will be filled in as they become
+available and prior to final upload into the rendered site]:
+
+* Person One foo@email (Chair of the Committee)
```
Comment (andyfaff) [inline doc/source/dev/conduct/enforcement.md]: Will publishing emails like this mean that if someone has an outdated scipy version (and outdated docs), then it's entirely possible that the whole committee could have changed?

File: doc/source/dev/conduct/enforcement.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
+# SciPy Code of Conduct - Enforcement Manual
```
Comment (andyfaff) [inline doc/source/dev/conduct/enforcement.md]: Enforcement manual sounds a little scary to me. `Enforcement Procedures`?

**Comment by pv** [comment]
Unimportant comment: the text as it is now is quite long and flowery. Personally, I would find something shorter to sound more sincere (c.f. [a traditional scandinavian approach](https://en.wikipedia.org/wiki/Christmas_Peace#The_text)).

**Comment by matthew-brett** [comment]
@pv - excellent - I think we should adopt the Scandinavian text as is :)

File: doc/source/dev/conduct/code_of_conduct.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
+# SciPy Code of Conduct
+
+SciPy is an engaged and respectful community made up of people
+from all over the world. Your involvement helps us to further our
+mission and to create an open platform that serves a broad range of
+communities, from research and education, to journalism, industry and
+beyond.
+
```
Comment (charris) [inline doc/source/dev/conduct/code_of_conduct.md]: A preamble stating the SciPY mission would be helpful here. I think the primary mission is to produce SciPy, that is,  software serving as a basis for the Python scientific stack, and that the aim of the code of conduct is to further that mission by keeping the people working on the project from fighting with each other and creating an unpleasant work environment.

File: doc/source/dev/conduct/code_of_conduct.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+Naturally, this implies diversity of ideas and perspectives on often complex
+problems. Disagreement and healthy discussion of conflicting viewpoints is
+welcome: the best solutions to hard problems rarely come from a single angle.
+But disagreement is not an excuse for aggression: humans tend to take
```
Comment (charris) [inline doc/source/dev/conduct/code_of_conduct.md]: Wandering off topic here. This is a CoC, not an essay on human nature.

File: doc/source/dev/conduct/code_of_conduct.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+disagreement personally and easily drift into behavior that ultimately degrades
+a community. This is particularly acute with online communication across
+language and cultural gaps, where many cues of human behavior are unavailable.
+We are outlining here a set of principles and processes to support a
+healthy community in the face of these challenges.
+
+Fundamentally, we are committed to fostering a productive, harassment-free
+environment for everyone. Rather than considering this code an exhaustive list
```
Comment (charris) [inline doc/source/dev/conduct/code_of_conduct.md]: "Harassment" belongs in the list of things not to do, not up here.

File: doc/source/dev/conduct/code_of_conduct.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+of things that you can’t do, take it in the spirit it is intended - a guide to
+make it easier to enrich all of us and the communities in which we participate.
+
+Importantly: as a member of our community, *you are also a steward of these
+values*.  Not all problems need to be resolved via formal processes, and often
+a quick, friendly but clear word on an online forum or in person can help
+resolve a misunderstanding and de-escalate things.
+
+However, sometimes these informal processes may be inadequate: they fail to
```
Comment (charris) [inline doc/source/dev/conduct/code_of_conduct.md]: Too wordy. Basically what you have here are methods to resolve bad behavior, which you have yet to specify. The desired behavior should come first, then perhaps a list of what constitutes "bad" behavior, and the various steps that can be taken to deal with the bad behavior.

File: doc/source/dev/conduct/code_of_conduct.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+   human to err: blaming each other doesn’t get us anywhere, while we can learn
+   from mistakes to find better solutions.
+
+8. **A simple apology can go a long way**. It can often de-escalate a situation,
+   and telling someone that you are sorry is an act of empathy that doesn’t
+   automatically imply an admission of guilt.
+
+
+## Reporting
+
```
Comment (charris) [inline doc/source/dev/conduct/code_of_conduct.md]: I think this should be considered a last resort. A formal report of bad behavior is likely to do much to disrupt the community and, for a small community such is SciPy, is pretty much a massive failure in itself.

File: doc/source/dev/conduct/code_of_conduct.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+If you believe someone is violating the code of conduct, please report this in
```
Comment (charris) [inline doc/source/dev/conduct/code_of_conduct.md]: "believe" <- "good reason"

File: doc/source/dev/conduct/code_of_conduct.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+a timely manner. Code of conduct violations reduce the value of the community
```
Comment (charris) [inline doc/source/dev/conduct/code_of_conduct.md]: "Code of conduct violations ..." is extraneous and can be omitted.

File: doc/source/dev/conduct/code_of_conduct.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+for everyone and we take them seriously.
+
+You can file a report by emailing *TODO: email address* or alternatively by contacting
+the NumFOCUS reporting contact (https://www.numfocus.org/about/code-of-conduct/).
+For more details, please see our [Reporting Guidelines](link).
+
+You can request that we follow up with you directly.  If you prefer to keep your
+report anonymous, please send it from an email account that isn't linked to your
```
Comment (charris) [inline doc/source/dev/conduct/code_of_conduct.md]: Anonymous reports are dangerous and subject to abuse. I think it would be better to assure confidentiality.

File: doc/source/dev/conduct/enforcement.md (hunk 4 above)
Comment (charris) [reply doc/source/dev/conduct/enforcement.md]: This is getting repetitious and sounds more like setting up a secret police department. What can we do beyond at most banning someone from the mailing lists and, perhaps, the github repos. Given such limited consequence I think the enforcement machinery can be simplified. More severe consequences start to impinge on the proper role of the court system.

File: doc/source/dev/conduct/code_of_conduct.md (hunk 12 above)
Comment (charris) [reply doc/source/dev/conduct/code_of_conduct.md]: It is also the case that most of SciPy related interactions take place in public, on the mailing lists or in the comments on pull requests. As everything is pretty much visible to everyone, I don't see the value in anonymous reporting. It might be worth emphasizing the value of keeping everything in the public sphere and listing precisely where the CoC is in effect. I don't think we want to get into policing the private interactions of people, a situation where everything immediately becomes far more difficult.

**Comment by rgommers** [comment]
The majority opinion seems to be against starting from the Jupyter CoC, both because it's too long and because it has various issues in both content and the way at which it was introduced for Jupyter itself. So I'm closing this PR and will open an issue that a CoC needs to be in place for 1.0.

Thanks for your thoughts everyone.
